project/src/main/webapp/shared.js

```diff
@@ -181,10 +181,15 @@ function togglePrefilledInterests() {
 }
 
 async function getRankedEvents(events) {
-  const response = await fetch('/event-ranker?' + new URLSearchParams({'events': JSON.stringify(events)}));
+  const eventIds = new Array(); 
+  for (index in events) {
```

Comment:
Use the for/of loop instead, for things that are arrays 
https://google.github.io/styleguide/jsguide.html#features-for-loops

Reply:
yes ty i was looking for something like that!